src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java

```diff
@@ -276,7 +276,7 @@ public Cigar getCigar() {
      */
     @Override
     public int numCigarElements(){
-        return samRecord.getCigar() == null ? 0 : samRecord.getCigar().numCigarElements();
+        return samRecord.getCigar() == null ? 0 : samRecord.getCigarLength();
```

Comment:
Sorry, you were right, it looks like samRecord.getCigarLength can null ptr if there is a null cigar since it just call `getCigar` internally.